src/Service/SignalsService.php

```diff
@@ -0,0 +1,264 @@
+                        "value"=> "bgp"
+                    ]
+                ]
+            ],
+            // NOTE: "ucsd-nt" migrated to using influxdb
+            "ucsd-nt" => [
+                "type" => "function",
+                "func" => "alias",
+                "args" => [
+                    [
+                        "type" => "path",
+                        "path" => sprintf("darknet.ucsd-nt.non-erratic.%s.uniq_src_ip", $fqid)
+                    ],
+                    [
+                        "type" => "constant",
+                        "value" => "ucsd-nt"
+                    ]
+                ]
+            ],
+            "ping-slash24" => [
+                "type" => "function",
+                "func" => "alias",
+                "args" => [
+                    [
+                        "type" => "function",
+                        "func" => "sumSeries",
+                        "args" => [
+                            [
+                                "type" => "function",
+                                "func" => "keepLastValue",
+                                "args" => [
+                                    [
+                                        "type" => "path",
+                                        "path" => "active.ping-slash24.geo.netacuity.NA.KN.probers.team-1.caida-sdsc.*.up_slash24_cnt"
+                                    ], [
+                                        "type" => "constant",
+                                        "value" => 1
+                                    ]
+                                ]
+                            ]
+                        ]
+                    ],
+                    [
+                        "type" => "constant",
+                        "value" => "ping-slash24"
+                    ]
+                ]
+            ],
+        ];
+
+        return $queryJsons[$datasource->getDatasource()];
+    }
+
+    /**
+     * Build influx query string based on the entity type and code.
+     *
+     * @param DatasourceEntity $datasourceEntity
+     * @param MetadataEntity $entity
+     * @param QueryTime $from
+     * @param QueryTime $until
+     * @param int $step
+     * @return string influx query string
+     * @throws BackendException
+     */
+    private function buildInfluxQuery(DatasourceEntity $datasourceEntity, MetadataEntity $entity,
+                                      QueryTime $from, QueryTime $until, int $step): string {
+        $entityType = $entity->getType()->getType();
+        $entityCode = $entity->getCode();
+        if($datasourceEntity->getDatasource() == "ucsd-nt"){
+            // create query template
+            if($entityType == "country"){
+                $template = "q=SELECT mean(\"uniq_src_ip\") FROM \"geo_country\" WHERE (\"telescope\" = 'ucsd-nt' AND \"country_code\" = '%s' AND \"filter\" = 'non-erratic' AND \"geo_db\" = 'netacuity') AND %s GROUP BY time(%ds) fill(null)";
```

Comment:
now i'm convinced that it's dangerous to have these full queries copy/pasted here. you've used `country_code` in the where clause for region and county.

i think this code should do something more like:

```
$MEASUREMENTS = [
  "country" => "geo_country",
  "region" => "geo_region",
  "county" => "geo_county",
  "asn" => "origin_asn",
];
$WHERE = [
  "country" => "country_code = %s AND geo_db = netacuity",
  "region" => "region_code = %s AND geo_db = netacuity",
  "county" => "county_code = %s AND geo_db = netacuity",
  "asn" => "asn = %s",
]; // of course with correct quoting..

$measurement = $MEASUREMENT[$entityType];
$where = sprintf($WHERE[$entityType], $entityCode);
$timeQuery = // what you already have
$step = // input parameter

$query = "q=SELECT mean(\"uniq_src_ip\") FROM \"$measurement\" WHERE (\"telescope\" = 'ucsd-nt' AND \"filter\" = 'non-erratic' AND $where) AND $timeQuery GROUP BY time($step) fill(null)";
```
(something like this anyway)

Reply:
addressed in here https://github.com/CAIDA/ioda-api/pull/3/commits/01c6aaa8568630b2f18620cd49732d68727503ea